test/tests_amdgpu.jl

```diff
@@ -91,6 +91,7 @@ end
 @testset "zeros-N" begin
 
 	N = 10
+	JACC.default_eltype() = Float32
```

Comment:
Should this be a setter function? Like `set_default_eltype(Float32)`

Reply:
This is not exactly possible in Julia. 
```
## In JACC/array.jl
function default_eltype()
	Float64
end

function set_default_eltype(T::Type)
	@show T
	@eval default_eltype() = T
end

## in command line 
julia> JACC.default_eltype()
Float64
julia> JACC.set_default_eltype(Float32)
T = Float32
julia> JACC.default_eltype()
ERROR: UndefVarError: `T` not defined
Stacktrace:
 [1] default_eltype()
   @ JACC ~/.julia/dev/JACC.jl/src/array.jl:9
 [2] top-level scope
   @ REPL[28]:1
```
I believe what is canonically done in Julia is local function overloading like I have done.  Though it would be possible to make a `DefaultEltype{T}` struct where T is the default eltype. Then 
```
struct DefaultEltype{T}
end
DefaultObject = DefaultEltype{Float64}()

default_eltype(::DefaultEltype{T}) where {T} = T
function default_eltype()
	return default_eltype(DefaultObject)
end

function set_default_eltype(T::Type)
	JACC.DefaultObject = DefaultEltype{T}()
end
```

Another code pattern which might be more attractive is
```
JACC.default_eltype(::Type{<:AbstractArray}) = Float64
## in CUDA extension
JACC.default_eltype(::Type{<:CuArray}) = Float32
```

Reply:
That's what I am afraid of. The non-portable signatures add more functionality. I am more tempted to just remove it to keep it simple for our targeted users. Thoughts?

Reply:
Correct, but the Julia ecosystem itself does not maintain consistency. My two cents for what's important for this effort is what would make integration more simple and less ambiguous. We are adding more code to support non-portable uses, I'd rather not buy ourselves into maintaining more code that does not add value, in terms of solving problems to potential users.